### PR TITLE
feat: introduce i18n support for side nav flow counterpart

### LINKED ISSUE #5203: Add i18n API to Side Nav Flow counterpart
### Describe your motivation

The i18n support was recently introduced for the web component. The Flow counterpart of the API is currently missing.

See: the web component [issue](https://github.com/vaadin/web-components/issues/5813)

### Describe the solution you'd like

The Flow counterpart for Side Nav should also provide i18n API.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ugur-vaadin**
## Description

This PR introduces i18n support for Side Nav. Currently, there is only a single property `toggle`, which is used for accessibility purposes in the web component.

See: related web component [PR](https://github.com/vaadin/web-components/pull/6014)

Closes #5203 

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.
- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java [hunk 1]
```diff
@@ -151,6 +162,65 @@ protected void onAttach(AttachEvent attachEvent) {
...
+     */
+    public void setI18n(SideNavI18n i18n) {
+        Objects.requireNonNull(i18n,
+                "The i18N properties object should not be null");
+        this.i18n = i18n;
+        requestI18nUpdate();
+    }
+
+    private void requestI18nUpdate() {
+        getUI().ifPresent(ui -> {
```
Comment (web-padawan) [inline vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java:195]: This logic seems to be copied from the `DatePicker` which has extra complexity.
Consider using a simpler approach like in `Avatar` and `AvatarGroup`:

```java
getElement().setPropertyJson("i18n", JsonSerializer.toJson(i18n));
```

Comment (ugur-vaadin) [reply vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java]: Yes, I checked a few implementations and combined some of their approaches. I simplified the approach with the latest commit. However, there are lots of reused logic regarding i18n, we can maybe consider introducing an interface like `hasInternationalization` in the future.

Comment (web-padawan) [reply vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java]: Makes sense, thank you. Let's consider this when we get to improving `i18n` (which would be a separate epic).

**Review by web-padawan — APPROVED** [review]

File: vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java [hunk 2]
```diff
@@ -42,6 +47,10 @@ public class SideNav extends SideNavItemContainer implements HasSize, HasStyle {
 
     private Element labelElement;
 
+    private SideNavI18n i18n;
+
+    private StateTree.ExecutionRegistration pendingI18nUpdate;
```
Comment (sissbruecker) [inline vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java:52]: Seems to be unused

File: vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java [hunk 3]
```diff
@@ -180,4 +218,31 @@ protected FeatureFlags getFeatureFlags() {
...
     }
 
+    /**
+     * The internationalization properties for {@link SideNav}.
+     */
+    public static class SideNavI18n implements Serializable {
+        private String toggle;
+
+        /**
+         * Gets the translated expression for toggling child items.
```
Comment (sissbruecker) [inline vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java:228]:
```suggestion
         * The text announced by screen readers when focusing the button for toggling child items.
```

File: vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java [hunk 4]
```diff
@@ -180,4 +218,31 @@ protected FeatureFlags getFeatureFlags() {
...
+         *
+         * @return the translated expression for toggling child items
+         */
+        public String getToggle() {
+            return toggle;
+        }
+
+        /**
+         * Sets the translated expression for toggling child items.
```
Comment (sissbruecker) [inline vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java:237]:
```suggestion
         * Sets the text announced by screen readers when focusing the button for toggling child items.
```

File: vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java (hunk 2 above)
Comment (ugur-vaadin) [reply vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java]: Removed.

**Pull request merged**